presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkFailureUtils.java

```diff
@@ -49,6 +49,7 @@ public static PrestoSparkFailure toPrestoSparkFailure(Session session, Execution
         requireNonNull(executionFailureInfo, "executionFailureInfo is null");
         PrestoSparkFailure prestoSparkFailure = toPrestoSparkFailure(executionFailureInfo);
         checkState(prestoSparkFailure != null);
+        checkState(prestoSparkFailure.getErrorCode() != null);
```

Comment:
It is not clear how this change would fix a test. Would you clarify?

Reply:
This change was not required. I updated the diff to fix where the NPE was coming from.